### PR TITLE
Use SubMonitor in udig.project and udig.project.ui

### PR DESCRIPTION
**Pull request by sschulz92**
With https://github.com/locationtech/udig-platform/pull/590 and this PR, I managed to get all `SubProgressMonitor` places to be updated :)

image

I assume this part to be done with this PR beeing merged.

### CODE REVIEW

File: plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/command/provider/FIDFeatureProvider.java [hunk 1]
```diff
@@ -54,11 +54,11 @@ public synchronized SimpleFeature get(IProgressMonitor monitor, Object... params
                 monitor = new NullProgressMonitor();
             }
             try {
-                monitor.beginTask("Get Feature", 100);
+                monitor.beginTask("Get Feature", 100); //$NON-NLS-1$
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/command/provider/FIDFeatureProvider.java:57]: this and maybe other Strings are potentially a message that is visible in UI and should be externailzed to allow translations. This is not part of this pull request, but once marked as `$NON-NLS-1$` its hard to find the messages which needs to be exported in the future ;)

See Transifex project to invoke translation community : https://www.transifex.com/udig/udig/dashboard/

File: plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/internal/MapEditor.java [hunk 2]
```diff
@@ -353,7 +349,7 @@ public boolean handlePreShutdownException(final Throwable t, final boolean force
         public boolean preShutdown(final IProgressMonitor monitor, final IWorkbench workbench,
                 final boolean forced) throws Exception {
             monitor.beginTask("Saving Map Editor", 3); //$NON-NLS-1$
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/internal/MapEditor.java:351]: same $NON-NLS-1$ issue here

File: plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/internal/MapEditorWithPalette.java [hunk 3]
```diff
@@ -381,7 +381,7 @@ public boolean handlePreShutdownException(Throwable t, boolean forced) {
         public boolean preShutdown(IProgressMonitor monitor, IWorkbench workbench, boolean forced)
                 throws Exception {
             monitor.beginTask("Saving Map Editor", 3); //$NON-NLS-1$
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/internal/MapEditorWithPalette.java:383]: again, $NON-NLS-1$ issue, please remove `// $NON-NLS-1$` here

File: plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/operations/MakeHole.java [hunk 4]
```diff
@@ -37,75 +37,81 @@
...
             }
 
+            @Override
             public Command copy() {
                 return this;
             }
 
+            @Override
             public String getName() {
                 return "Create Hole Command"; //$NON-NLS-1$
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/operations/MakeHole.java:110]: please remove `//$NON-NLS-1$` here too

**Review by fgdrf — CHANGES_REQUESTED** [review]
Migration from SubProgressMonitor to SubMonitor looks good.

Can you please remove the `$NON-NLS-1$` comments for messages that are potentially shown in the UI and might be exported to properties files ( to allow translations)?

Thanks for removing depprecations with this changeset!

**Comment by sschulz92** [comment]
May I ask: how long are those strings there? Do you think this kind of change will be done sooner than later?

In Eclipse warnings are shown which I want to get rid of. I would prefer TODO comments, to externalize those strings. What do you think?

**Comment by fgdrf** [comment]
> 
> 
> May I ask: how long are those strings there? Do you think this kind of change will be done sooner than later?
> 
> In Eclipse warnings are shown which I want to get rid of. I would prefer TODO comments, to externalize those strings. What do you think?

Hmm, thats why this are warnings in the IDE, I guess no need to add aditional TODO right there. I guess the challange is to identify those messages to export and those that are not visible in UI

**Review by fgdrf — APPROVED** [review]
waiting for feedback from github workflow ..

**Pull request merged**